development/development/css_selectors.rst

```diff
@@ -0,0 +1,256 @@
+
+.. code:: css
+
+    input.btn { }
+
+This is a *qualified* selector; the leading ``input`` ties this ruleset to only being able to work on ``input`` elements. By omitting this qualification, we allow ourselves to reuse the ``.btn`` class on any element we choose, like an ``a``, for example, or a ``button``.
+
+Qualified selectors do not lend themselves well to being reused, and every selector we write should be authored with reuse in mind.
+
+Of course, there are times when you may want to legitimately qualify a selector—you might need to apply some very specific styling to a particular element when it carries a certain class, for example:
+
+.. code:: css
+
+    /**
+     * Embolden and color any element with a class of `.error`.
+     */
+    .error {
+        color: red;
+        font-weight: bold;
+    }
+
+    /**
+     * If the element is a `div`, also give it some box-like styling.
+     */
+    div.error {
+        padding: 10px;
+        border: 1px solid;
+    }
+
+This is one example where a qualified selector might be justifiable, but I would still recommend an approach more like:
+
+.. code:: css
+
+    /**
+     * Text-level errors.
+     */
+    .error-text {
+        color: red;
+        font-weight: bold;
+    }
+
+    /**
+     * Elements that contain errors.
+     */
+    .error-box {
+        padding: 10px;
+        border: 1px solid;
+    }
+
+This means that we can apply ``.error-box`` to any element, and not just a ``div``—it is more reusable than a qualified selector.
+
+Quasi-Qualified Selectors
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+One thing that qualified selectors can be useful for is signaling where a class might be expected or intended to be used, for example:
+
+.. code:: css
+
+    ul.nav { }
+
+Here we can see that the ``.nav`` class is meant to be used on a ``ul`` element, and not on a ``nav``. By using *quasi-qualified selectors* we can still provide that information without actually qualifying the selector:
+
+.. code:: css
+
+    /*ul*/.nav { }
+
+By commenting out the leading element, we can still leave it to be read, but avoid qualifying and increasing the specificity of the selector.
+
+Naming
+~~~~~~
+
+As Phil Karlton once said
+
+    ‘There are only two hard things in Computer Science: cache
+    invalidation and naming things.’
+
+I won’t comment on the former claim here, but the latter has plagued me for years. My advice with regard to naming things in CSS is to pick a name that is sensible, but somewhat ambiguous: aim for high reusability. For example, instead of a class like ``.site-nav``, choose something like ``.primary-nav``; rather than ``.footer-links``, favor a class like ``.sub-links``.
```

Comment:
Same here, probably better to get rid of I as this documentation is seen as the phpBB documentation and not by a specific person.